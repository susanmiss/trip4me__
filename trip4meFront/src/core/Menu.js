import React from 'react'
import { Link, withRouter } from 'react-router-dom'
import { signout, isAuthenticated, signin } from '../auth/index.js'


const isActive = (history, path) => {
  if (history.location.pathname === path) return { color: "purple" }
  else return { color: "#ffffff" }
}

const Menu = ({ history }) => (

  <nav className="navbar navbar-expand-lg navbar-dark fixed-top navbar-shrink" id="mainNav" data-test="nav-element">

    <div className="container">
      <h4>
        <Link className="navbar-brand js-scroll-trigger" to="/">
          Trip4me
            </Link>
      </h4>

      <button className="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">

        <i className="fas fa-bars"></i>
      </button>

      <div className="collapse navbar-collapse" id="navbarResponsive">
        <ul className="navbar-nav text-uppercase ml-auto">



          {!isAuthenticated() && (
            <>
              {/* <li className="nav-item">
                <Link className="nav-link js-scroll-trigger" to="/posts">All Places</Link>
              </li> */}

              <li className="nav-item">
                <Link className="nav-link js-scroll-trigger" to="/signin">Sign In</Link>
              </li>

              <li className="nav-item">
                <a className="nav-link js-scroll-trigger" href="/#contact">Contact</a>
              </li>
            </>
          )}
        </ul>

        {isAuthenticated() && (
          <>
            <ul className="navbar-nav text-uppercase ml-auto">
              <li className="nav-item">
                <Link className="nav-link js-scroll-trigger" to={`/category/create`}>
                  Create Category
                      </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link js-scroll-trigger" to={`/post/new`}>
                  Create Post
                      </Link>
              </li>
              {/* <li className="nav-item">
                <Link
                  to={`/admin`}
                  className="nav-link">
                  Admin
                </Link>
              </li> */}
              <li className="nav-item">
                <Link className="nav-link js-scroll-trigger" onClick={() => signout(() => history.push('/'))} to="/">Sign Out</Link>
              </li>
            </ul>
          </>
        )}
      </div>
    </div>
  </nav>
)

export default withRouter(Menu);
