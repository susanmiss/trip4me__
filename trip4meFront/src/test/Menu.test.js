import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Menu from '../core/Menu'


const setup = (props={}, state=null) => {
    const wrapper = shallow(<Menu {...props} />)
    if( state ) wrapper.setState(state);
    return wrapper;
  }

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

describe("<Menu />", () => {

  let wrapper;
  beforeEach(() => wrapper = shallow(<Menu />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  

})