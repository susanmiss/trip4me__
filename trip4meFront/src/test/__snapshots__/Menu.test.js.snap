// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Renders correctly 1`] = `ShallowWrapper {}`;
